crates/utils/src/format/json.rs

```diff
@@ -0,0 +1,114 @@
+#![cfg(feature = "serde")]
+
+use js_sys::JsString;
+use wasm_bindgen::{prelude::wasm_bindgen, JsValue, UnwrapThrowExt};
+mod private {
+    pub trait Sealed {}
+    impl Sealed for wasm_bindgen::JsValue {}
+}
+
+// Turns out `JSON.stringify(undefined) === undefined`, so if
+// we're passed `undefined` reinterpret it as `null` for JSON
+// purposes.
+#[wasm_bindgen(
+    inline_js = "export function serialize(obj) { return JSON.stringify(obj === undefined ? null : obj) }"
```

Comment:
Inline JS snippets could cause issues for non-ES module targets.
`nomodule` target is still the only target supported by `gloo-worker`.

See: https://rustwasm.github.io/docs/wasm-bindgen/examples/without-a-bundler.html#using-the-older---target-no-modules

Instead of going through `JSON::stringify` and `serde_json`, I would suggest using [`serde-wasm-bindgen`](https://docs.rs/serde-wasm-bindgen/latest/serde_wasm_bindgen/). It supports non-serializable types (e.g.: ES2015 `Map`) and has a smaller bundle footprint than `serde_json`.

Reply:
@futursolo oh well, back to what I had before.

So if you use [`serde_wasm_bindgen`](https://docs.rs/serde-wasm-bindgen/latest/serde_wasm_bindgen/), then you don't really need this. The goal is to provide a drop-in replacement for `JsValue::{from_serde/to_serde}` and an easier way to use `js_sys::JSON::`.

Plus, in the deprecation thread, [someone](https://github.com/rustwasm/wasm-bindgen/pull/3031#issuecomment-1216561512) said they had a bad time with `serde-wasm-bindgen`. I think it makes sense: it calls to JS for every field which is slow.